<a href="https://colab.research.google.com/github/mallenarenas/Distribuciones-de-Probabilidad/blob/main/clase_intervalos_de_confianza.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from scipy.stats import norm, t, chi2, f
import statistics as stat
import math

 <h1>Ejemplo 1 </h1>

 <p> Suponga que una máquina envasadora de café debe envasar este producto en tarros de 500 gr. Para verificar esto se toma una muestra de 16 tarros, arrojando los siguientes resultados: </p>    

In [ ]:
x=[497, 512, 504, 510, 506, 508, 503, 499, 502, 496, 509, 514, 593, 505, 496, 506]

Si el peso de cada tarro está normalmente distribuido con $\sigma=5$ gr, obtenga el intervalo de confianza para la media de llenado con una confianza del 90\% y una del 95\%.
    

Al 90% de confianza: \\
$\gamma = 0.9$ \\
$\alpha = 0.1$ \\
$\frac{\alpha}{2}=0.05$ \\

In [ ]:
norm.ppf(0.95,0,1)

1.6448536269514722

In [ ]:
norm.ppf(0.05,0,1)

-1.6448536269514729

In [ ]:
stat.mean(x)

510

$\overline{X}=510$ \\
$\sigma =5$ \\
$n=16$ \\
$\sqrt{n}=\sqrt{16}=4$ \\

$$IC(\mu) = (\overline{X} - z_{1 - \alpha/2}\frac{\sigma}{\sqrt{n}}, \overline{X} + z_{1 - \alpha/2}\frac{\sigma}{\sqrt{n}}).$$
$$= (510 - 1.645\frac{5}{4}, 510 + 1.645\frac{5}{4})$$

In [ ]:
limite_inferior = 510 - norm.ppf(0.95,0,1)*5/4
limite_superior = 510 + norm.ppf(0.95,0,1)*5/4
print(f"({limite_inferior},{limite_superior})")

(507.9439329663107,512.0560670336894)


Al 95% de confianza: \\
$\gamma = 0.95$ \\
$\alpha = 0.05$ \\
$\frac{\alpha}{2}=0.025$ \\

In [ ]:
norm.ppf(0.975,0,1)

1.959963984540054

$$IC(\mu) = (\overline{X} - z_{1 - \alpha/2}\frac{\sigma}{\sqrt{n}}, \overline{X} + z_{1 - \alpha/2}\frac{\sigma}{\sqrt{n}})$$
$$= (510 - 1.96\frac{5}{4}, 510 + 1.96\frac{5}{4})$$

In [ ]:
limite_inferior = 510 - norm.ppf(0.975,0,1)*5/4
limite_superior = 510 + norm.ppf(0.975,0,1)*5/4
print(f"({limite_inferior},{limite_superior})")

(507.55004501932495,512.4499549806751)


 En el ejemplo anterior, ¿Cuál deberá ser el tamaño muestral para que con un 90\% de confianza tengamos un error de estimación menor a 1 gramo?

In [ ]:
(norm.ppf(0.95,0,1)*5)**2

67.63858635238533

  <h1>Ejemplo 2 </h1>

 Se afirma que una empresa de artículos de escritorio produce cuadernos de tal manera que 1 de cada 4 cuadernos son defectuosos. Para comprobar esta afirmación se toma una muestra de 250 cuadernos, de los cuales 25 resultan ser defectuosos. Determinar el intervalo del 95\% de confianza para la verdadera proporción de defectuosos producidos por la empresa.

$$IC(p) = (\widehat{p} - z_{1 - \alpha/2}\sqrt{\frac{\widehat{p}(1-\widehat{p})}{n}}, \widehat{p} + z_{1 - \alpha/2}\sqrt{\frac{\widehat{p}(1-\widehat{p})}{n}})$$
$$= (1/10 - 1.96\sqrt{\frac{\frac{1}{10}\frac{9}{10}}{250}}, 1/10 + 1.96\sqrt{\frac{\frac{1}{10}\frac{9}{10}}{250}})$$

In [ ]:
limite_inferior = 1/10 - norm.ppf(0.975,0,1)*math.sqrt((1/10*9/10)/250)
limite_superior = 1/10 + norm.ppf(0.975,0,1)*math.sqrt((1/10*9/10)/250)
print(f"({limite_inferior},{limite_superior})")

(0.06281229806172631,0.1371877019382737)


  <h1>Ejemplo 3 </h1>

  Considere los puntajes de una prueba de estadística como una variable
aleatoria. Se toma una muestra aleatoria de 9 alumnos con los siguientes
resultados: 80, 8, 20, 65, 50, 30, 31, 85, 31.
Considerando que estos puntajes tienen distribución de probabilidad
normal, determinar el intervalo de confianza de 90% para la media de la
población

In [ ]:
t.ppf(0.95,8)

In [ ]:
x=[80, 8, 20, 65, 50, 30, 31, 85, 31]
media=stat.mean(x)
print("media:", media)
S = stat.stdev(x)
print("S:",S)

media: 44.44444444444444
S: 27.060631511067474


$$IC(\mu) = (\overline{X} - t_{(1 - \alpha/2,n-1)}\frac{S}{\sqrt{n}}, \overline{X} + t_{(1 - \alpha/2,n-1)}\frac{S}{\sqrt{n}})$$
$$= (44.44 - 1.86\frac{27.06}{3}, 44.44 + 1.86\frac{27.06}{3})$$

In [ ]:

limite_inferior = media - t.ppf(0.95,8)*S/3
limite_superior = media + t.ppf(0.95,8)*S/3
print(f"({limite_inferior},{limite_superior})")

(27.67092970426634,61.21795918462254)


In [ ]:
cuadrados_de_distancia = []
for i in range(0,len(x)):
  cuadrados_de_distancia.append((x[i] - media)**2)
print(math.sqrt((sum(cuadrados_de_distancia)/8)))

27.060631511067474


 <h1>Ejemplo 4</h1>
 <h2>varianzas conocidas</h2>
Suponga que se tienen dos materiales de construcción hechos con un
componente $A$ y otro con un componente $B$. Se sabe que las poblaciones
de ambos componentes poseen distribución normal con varianzas $\sigma_A^2=16$
y $\sigma_B^2=4$ para los componentes $A$ y $B$ respectivamente. El objetivo de este
estudio consiste en medir las resistencias de ambos materiales para elegir
cuál es mejor. Para esto se midieron las resistencias (en kg/cm2) de los
componentes obteniéndose los siguientes resultados:

In [ ]:
componente_A = [31, 34, 32, 35, 37, 32, 42]
componente_B = [40, 38, 35, 42, 41, 39, 40, 38, 37]

In [ ]:
X_barra_A = stat.mean(componente_A)
print("media de A:", X_barra_A)
X_barra_B = stat.mean(componente_B)
print("media de B:",X_barra_B)

media de A: 34.714285714285715
media de B: 38.888888888888886


Como no se da un nivel de confianza lo fijamos en $\gamma = 0.95$

luego $1-\alpha/2 = 0.975$

In [ ]:
norm.ppf(0.975,0,1)

1.959963984540054

$$IC(\mu_1-\mu_2)=(\overline{X_1}-\overline{X_2})\pm z_{1-\alpha/2}\sqrt{\frac{\sigma_1^2}{n_1}+\frac{\sigma_2^2}{n_2}}$$
$$=(38.88-34.71)\pm 1.96\sqrt{\frac{16}{7}+\frac{4}{9}}.$$

In [ ]:
limite_inferior = (X_barra_B - X_barra_A) - norm.ppf(0.975,0,1)*math.sqrt((16/7)+(4/9))
limite_superior = (X_barra_B - X_barra_A) + norm.ppf(0.975,0,1)*math.sqrt((16/7)+(4/9))
print(f"({limite_inferior},{limite_superior})")

(0.9361170566981816,7.413089292508159)


 <h1>Ejemplo 5 </h1>

  <h2>varianzas desconocidas iguales</h2>

 Para comparar el efecto del estrés en la forma de ruido sobre la capacidad de realizar un trabajo sencillo,
70 personas fueron divididas en dos grupos. El primer grupo de 30 personas actuó como control, en tanto que
el segundo grupo de 40 fueron el grupo experimental. Aun cuando cada persona realizó
el trabajo en el mismo cuarto de control, cada una de las personas del grupo experimental tuvo que
realizar el trabajo cuando se reproducía música de rock a alto volumen. El tiempo para terminar el trabajo
se registró para cada individuo y se obtuvo el siguiente resumen:

In [ ]:
n_control = 30
n_experimental = 40
x_barra_control = 15
x_barra_experimental = 23
S_control = 4
S_experimental = 10

a) Encuentre un intervalo de confianza de 99\% para la diferencia en tiempos medios de terminación
para estos dos grupos.

b) Con base en el intervalo de confianza del inciso a),
   ¿hay suficiente evidencia para indicar una diferencia en el tiempo promedio de terminación para los dos grupos? Explique.


$$S_p^2=\frac{(n_1-1)S_1^2+(n_2-1)S_2^2}{n_1+n_2-2}.$$

In [ ]:
S_p = math.sqrt(((n_control -1)*S_control**2+(n_experimental -1)*S_experimental**2)/(n_control + n_experimental -2))
S_p

8.011021819233505

$$IC(\mu_1-\mu_2)=(\overline{X_1}-\overline{X_2})\pm t_{1-\alpha/2,n_1+n_2-2}S_p\sqrt{\frac{1}{n_1}+\frac{1}{n_2}},$$


In [ ]:
limite_inferior = (x_barra_experimental-x_barra_control) - t.ppf(0.995,n_control+n_experimental-2) * S_p * math.sqrt(1/n_experimental+1/n_control)
limite_superior = (x_barra_experimental-x_barra_control) + t.ppf(0.995,n_control+n_experimental-2) * S_p * math.sqrt(1/n_experimental+1/n_control)
print("(",limite_inferior,limite_superior,")")

( 2.872501901671577 13.127498098328424 )


 <h1>Ejemplo 6 </h1>

  <h2>varianzas desconocidas distintas</h2>

 Para comparar el efecto del estrés en la forma de ruido sobre la capacidad de realizar un trabajo sencillo,
70 personas fueron divididas en dos grupos. El primer grupo de 30 personas actuó como control, en tanto que
el segundo grupo de 40 fueron el grupo experimental. Aun cuando cada persona realizó
el trabajo en el mismo cuarto de control, cada una de las personas del grupo experimental tuvo que
realizar el trabajo cuando se reproducía música de rock a alto volumen. El tiempo para terminar el trabajo
se registró para cada individuo y se obtuvo el siguiente resumen:

In [ ]:
n_control = 30
n_experimental = 40
x_barra_control = 15
x_barra_experimental = 23
S_control = 4
S_experimental = 10

a) Encuentre un intervalo de confianza de 99\% para la diferencia en tiempos medios de terminación
para estos dos grupos.

b) Con base en el intervalo de confianza del inciso a),
   ¿hay suficiente evidencia para indicar una diferencia en el tiempo promedio de terminación para los dos grupos? Explique.

In [ ]:
nu = (S_control**2/n_control + S_experimental**2/n_experimental)**2/((S_control**2/n_control)**2*1/(n_control-1)+(S_experimental**2/n_experimental)**2*1/(n_experimental-1))-2
nu

52.1035474758678

In [ ]:
limite_inferior = (x_barra_control-x_barra_experimental)-t.ppf(0.995,nu)*math.sqrt(S_control**2/n_control+S_experimental**2/n_experimental)
limite_superior = (x_barra_control-x_barra_experimental)+t.ppf(0.995,nu)*math.sqrt(S_control**2/n_control+S_experimental**2/n_experimental)
print(limite_inferior, limite_superior)

-12.656348664909977 -3.3436513350900237


 <h1>Ejemplo 7 </h1>

  <h2>Muestras Pareadas</h2>

Suponga que se han implementado 2 m ́etodos para realizar un trabajo y se quiere saber si alguno de estos dos m ́etodos es m ́as eficiente en cuanto a que emplea menos tiempo. Para esto se tomaron 7 trabajadores y se les tomo ́ el tiempo (en minutos) que demoraban en realizar el trabajo utilizando, cada trabajador, ambos m ́etodos.
La siguiente tabla muestra los resultados obtenidos:

In [ ]:
x_A = [16, 14, 21, 23, 18, 13, 25]
x_B = [14, 15, 19, 24, 20, 15, 28]

x_A_barra = stat.mean(x_A)
x_B_barra = stat.mean(x_B)
d_barra = x_A_barra - x_B_barra

print(x_A_barra, x_B_barra, d_barra)

18.571428571428573 19.285714285714285 -0.7142857142857117


In [ ]:
d = []
for i in range(len(x_A)):
  d.append(x_A[i]-x_B[i])

S_d = stat.stdev(d)
S_d


1.9760470401187074

In [ ]:
limite_inferior = d_barra - t.ppf(0.975, 6)*S_d/math.sqrt(7)
limite_superior = d_barra + t.ppf(0.975, 6)*S_d/math.sqrt(7)
print("(",limite_inferior,",", limite_superior, ")")

( -2.5418244160583336 , 1.1132529874869104 )


Realice el análisis bajo un 95% de confianza.

 <h1>Ejemplo 8 </h1>

  <h2>diferencia de proporciones</h2>

In [ ]:
p_1 = 25/250
p_2 = 14/180

n_1 = 250
n_2 = 180


In [ ]:
limite_inferior = (p_1 - p_2) - norm.ppf(0.995,0,1)*math.sqrt((p_1*(1-p_1)/n_1)+(p_2*(1-p_2)/n_2))
limite_superior = (p_1 - p_2) + norm.ppf(0.995,0,1)*math.sqrt((p_1*(1-p_1)/n_1)+(p_2*(1-p_2)/n_2))
print("(",limite_inferior,",", limite_superior, ")")

( -0.04871796323699808 , 0.09316240768144253 )


 <h1>Ejemplo 9 </h1>

  <h2>intervalo para la varianza</h2>

In [ ]:
x=[497, 512, 504, 510, 506, 508, 503, 499, 502, 496, 509, 514, 593, 505, 496, 506]

In [ ]:
n = len(x)
n

16

In [ ]:
S = stat.stdev(x)
print(S,S**2)

22.806431841332245 520.1333333333333


In [ ]:
limite_inferior = ((n-1)*S**2)/(chi2.ppf(0.975,n-1))
limite_superior = ((n-1)*S**2)/(chi2.ppf(0.025,n-1))
print("(",limite_inferior,",", limite_superior, ")")

( 283.8288887516571 , 1245.9004026030238 )


In [ ]:
print("(", math.sqrt(limite_inferior),",", math.sqrt(limite_superior), ")")

( 16.847221989148746 , 35.29731438230144 )


 <h1>Ejemplo 10 </h1>

  <h2>intervalo para razón de varianzas</h2>

In [ ]:
s_A = 25
s_B = 36
n_A = 30
n_B = 40

In [ ]:
limite_inferior = s_A**2/s_B**2*f.ppf(0.025,n_B-1,n_A-1)
limite_superior = s_A**2/s_B**2*f.ppf(0.975,n_B-1,n_A-1)
print("(",limite_inferior,",", limite_superior, ")")

( 0.24581310011127666 , 0.9802594159848366 )


In [ ]:
s_C = 4
s_E = 10
n_C = 30
n_E = 40

In [ ]:
limite_inferior = s_C**2/s_E**2*f.ppf(0.005,n_E-1,n_C-1)
limite_superior = s_C**2/s_E**2*f.ppf(0.995,n_E-1,n_C-1)
print("(",limite_inferior,",", limite_superior, ")")

( 0.06577428602094264 , 0.41034464275761345 )


In [ ]:
n = (norm.ppf(0.975,0,1)*(15000/5000))**2
n

34.573129386247125

In [ ]:
x_barra = (197.75+194.65)/2
print(197.75 - x_barra, 194.65 - x_barra)
print(x_barra)

print(1.55*3/1.8)

1.5500000000000114 -1.549999999999983
196.2
2.5833333333333335


In [ ]:
norm.ppf(0.995,0,1)

2.5758293035489004

PROBLEMA 4 Guía 2

In [ ]:
datos = [85,78,92,88,76,90,87,89,82,94,81,79,91,86, 83,77,84,88,75,93,80,85,89,87,82,91,88,96]
media = stat.mean(datos)
S = stat.stdev(datos)

In [ ]:
limite_inferior = media - t.ppf(0.95,27)*S/math.sqrt(28)
limite_superior = media + t.ppf(0.95,27)*S/math.sqrt(28)
print(f"({limite_inferior},{limite_superior})")

(83.74963363376635,87.39322350909079)


problema 15 guía 2

In [ ]:
t.ppf(0.95,5)*1.128/math.sqrt(6)
media = 18.72
S = 1.128

In [ ]:
limite_inferior = media - t.ppf(0.95,5)*S/math.sqrt(6)
limite_superior = media + t.ppf(0.95,5)*S/math.sqrt(6)
print(f"({20* limite_inferior},{20*limite_superior})")

(355.8412403965554,392.95875960344455)


In [ ]:
0.68*math.sqrt(6)/S

1.4766427527416321

In [ ]:
t.pdf(0.68*math.sqrt(6)/S,5)

0.12816946197337525

In [ ]:
n = 6

In [ ]:
limite_inferior = ((n-1)*S**2)/(chi2.ppf(0.95,n-1))
limite_superior = ((n-1)*S**2)/(chi2.ppf(0.05,n-1))
print("(",math.sqrt(limite_inferior),",", math.sqrt(limite_superior), ")")

( 0.7580721274314148 , 2.3566824182208284 )


Problema 8 guía 2

In [ ]:
norm.ppf(0.95,0,1)**2*1/4/0.1**2

67.6385863523853

Problema 14 guía 2

In [ ]:
x = [25,28,30,32,26,31,27,29,24,30,28,33,25,26,31]
y = [20,24,28,22,19,25,23,27,21,26,25,29,23,20,28]

In [ ]:
d=[]
for i in range(0,len(x)):
  d.append(x[i]-y[i])



[5, 4, 2, 10, 7, 6, 4, 2, 3, 4, 3, 4, 2, 6, 3]

In [ ]:
d_barra = stat.mean(d)
S_d = stat.stdev(d)

In [ ]:
limite_inferior = d_barra - t.ppf(0.95, 14)*S_d/math.sqrt(15)
limite_superior = d_barra + t.ppf(0.95, 14)*S_d/math.sqrt(15)
print("(",limite_inferior,",", limite_superior, ")")

( 3.335997910568926 , 5.33066875609774 )


Problema 16 guía 2

In [ ]:
s_A = 7
s_B = 9
n_A = 18
n_B = 22
x_barra_A = 12
x_barra_B = 15

In [ ]:
limite_inferior = s_A**2/s_B**2*f.ppf(0.05,n_B-1,n_A-1)
limite_superior = s_A**2/s_B**2*f.ppf(0.95,n_B-1,n_A-1)
print("(",limite_inferior,",", limite_superior, ")")

( 0.28283047253487315 , 1.3422966523532684 )


In [ ]:
S_p = math.sqrt(((n_A -1)*s_A**2+(n_B -1)*s_B**2)/(n_A + n_B -2))
S_p

8.166040076212937

In [ ]:
limite_inferior = (x_barra_A-x_barra_B) - t.ppf(0.975,n_A+n_B-2) * S_p * math.sqrt(1/n_B+1/n_A)
limite_superior = (x_barra_A-x_barra_B) + t.ppf(0.975,n_A+n_B-2) * S_p * math.sqrt(1/n_B+1/n_A)
print("(",limite_inferior,limite_superior,")")

( -8.25398690994842 2.253986909948421 )


In [ ]:
limite_inferior = 185.7 - norm.ppf(0.995,0,1)*2.5/math.sqrt(9)
limite_superior = 185.7 + norm.ppf(0.995,0,1)*2.5/math.sqrt(9)
print(f"({limite_inferior},{limite_superior})")

(183.5534755803759,187.84652441962407)


In [ ]:
(norm.ppf(0.995,0,1)*2.5)**2

41.46810375638258

In [ ]:
x = [12.5, 15.2, 14.6, 13.8, 11.9, 17.2, 12.7, 13.9, 14.5]
y = [10.2, 11.8, 9.5, 12.1, 10.8, 12.5, 9.9, 9.7, 12.2]
x_barra = stat.mean(x)
y_barra = stat.mean(y)
s_x = stat.stdev(x)
s_y = stat.stdev(y)
n_x = 9
n_y = 9

In [ ]:
S_p = math.sqrt(((n_x -1)*s_x**2+(n_y -1)*s_y**2)/(n_x + n_y -2))
S_p

1.4150971698084904

In [ ]:
limite_inferior = (x_barra-y_barra) - t.ppf(0.95,n_x+n_y-2) * S_p * math.sqrt(1/n_x+1/n_y)
limite_superior = (x_barra-y_barra) + t.ppf(0.95,n_x+n_y-2) * S_p * math.sqrt(1/n_x+1/n_y)
print("(",limite_inferior,limite_superior,")")

( 1.9020169914706924 4.231316341862641 )


In [ ]:
n_h = 1545
n_m = 1691
p_h = 0.675
p_m = 0.651

In [ ]:
limite_inferior = (p_h - p_m) - norm.ppf(0.975,0,1)*math.sqrt((p_h*(1-p_h)/n_h)+(p_m*(1-p_m)/n_m))
limite_superior = (p_h - p_m) + norm.ppf(0.975,0,1)*math.sqrt((p_h*(1-p_h)/n_h)+(p_m*(1-p_m)/n_m))
print("(",limite_inferior,",", limite_superior, ")")

( -0.008581892553179336 , 0.05658189255317938 )
